[BUG] Configuration error

### Has this bug been raised before?

- [x] I have checked "open" AND "closed" issues and this is not a duplicate

### Where did you find this bug?

Development environment 

### Description

/Users/duminduudara/Documents/Project/carrefour/.next/types/app/api/auth/[...nextauth]/route.ts
    9:9    error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
   10:10   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
   11:13   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
   12:10   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
   13:9    error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
   14:12   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
   15:11   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
   16:12   error  The `{}` ("empty object") type allows any non-nullish value, including literals like `0` and `""`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead  @typescript-eslint/no-empty-object-type
   17:26   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  305:72   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  305:88   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  308:19   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  310:12   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  311:18   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  316:12   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  321:51   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  324:37   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  324:42   error  '_M' is defined but never used                                                                                                                                                                                                                                                                                                                                                             @typescript-eslint/no-unused-vars
  325:84   error  The `{}` ("empty object") type allows any non-nullish value, including literals like `0` and `""`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead  @typescript-eslint/no-empty-object-type
  327:15   error  'T' is defined but never used                                                                                                                                                                                                                                                                                                                                                              @typescript-eslint/no-unused-vars
  327:25   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  327:67   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  327:76   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  327:102  error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  328:16   error  'T' is defined but never used                                                                                                                                                                                                                                                                                                                                                              @typescript-eslint/no-unused-vars
  328:26   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  328:59   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  328:77   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  328:103  error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  329:84   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  337:22   error  '_' is defined but never used                                                                                                                                                                                                                                                                                                                                                              @typescript-eslint/no-unused-vars
  337:46   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any

/Users/duminduudara/Documents/Project/carrefour/.next/types/app/layout.ts
   9:12   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  10:12   error  The `{}` ("empty object") type allows any non-nullish value, including literals like `0` and `""`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead  @typescript-eslint/no-empty-object-type
  11:26   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  20:14   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  21:22   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  22:14   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  23:22   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  45:72   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  45:88   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  48:19   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  50:12   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  51:18   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  56:12   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  61:51   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  64:37   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  64:42   error  '_M' is defined but never used                                                                                                                                                                                                                                                                                                                                                             @typescript-eslint/no-unused-vars
  65:84   error  The `{}` ("empty object") type allows any non-nullish value, including literals like `0` and `""`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead  @typescript-eslint/no-empty-object-type
  67:15   error  'T' is defined but never used                                                                                                                                                                                                                                                                                                                                                              @typescript-eslint/no-unused-vars
  67:25   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  67:67   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  67:76   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  67:102  error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  68:16   error  'T' is defined but never used                                                                                                                                                                                                                                                                                                                                                              @typescript-eslint/no-unused-vars
  68:26   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  68:59   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  68:77   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  68:103  error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any
  69:84   error  The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type                                                                                                                                                                                                                                                                    @typescript-eslint/no-unsafe-function-type
  73:22   error  '_' is defined but never used                                                                                                                                                                                                                                                                                                                                                              @typescript-eslint/no-unused-vars
  73:46   error  Unexpected any. Specify a different type                                                                                                                                                                                                                                                                                                                                                   @typescript-eslint/no-explicit-any

/Users/duminduudara/Documents/Project/carrefour/auth.js
  19:11  error  'process' is not defined  no-undef
  44:9   error  'process' is not defined  no-undef
  70:11  error  'process' is not defined  no-undef

/Users/duminduudara/Documents/Project/carrefour/components/modules/website/auth/Register.tsx
  35:75  warning  Unused eslint-disable directive (no problems were reported)

/Users/duminduudara/Documents/Project/carrefour/components/modules/website/order/StripeForm.jsx
  69:17  error  'process' is not defined  no-undef

/Users/duminduudara/Documents/Project/carrefour/components/ui/input.tsx
  5:18  error  An interface declaring no members is equivalent to its supertype  @typescript-eslint/no-empty-object-type

/Users/duminduudara/Documents/Project/carrefour/components/ui/skeleton.tsx
  8:5  error  'React' must be in scope when using JSX  react/react-in-jsx-scope

/Users/duminduudara/Documents/Project/carrefour/components/ui/table.tsx
  72:6  error  'className' is missing in props validation  react/prop-types
  87:6  error  'className' is missing in props validation  react/prop-types

/Users/duminduudara/Documents/Project/carrefour/hooks/usePagination.tsx
  4:51  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

✖ 5398 problems (5397 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

### Steps to Reproduce

If you know the steps, follow the below format and provide steps to reproduce

For example:

1. Go to <replace your> page.
1. On top right side, close to the <Account> menu
1. Some menu are not working properly.
1. Continue...

If you don't know exact steps, include any relevant details like:

- What page you were on...
- What you were trying to do...
- What went wrong...


### Screenshots

[


](<img width="1680" alt="Image" src="https://github.com/user-attachments/assets/4f619288-158e-4199-a406-1952a124902a" />)

### Do you want to work on this issue?

yes

### If "yes" to above, please explain how you would technically implement this

_No response_

--- a/eslint.config.mjs
+++ b/eslint.config.mjs
@@ -20,7 +20,7 @@ export default [
   ...tseslint.configs.recommended,
   pluginReact.configs.flat.recommended,
 
-  //ignore some folders
+  //ignore some folders and files from linting
   {
     ignores: [
       ".next/*",
@@ -32,3 +32,4 @@ export default [
     ],
   },
 ];
+

